app/src/main/java/space/taran/arknavigator/ui/fragments/SettingsFragment.kt

```diff
@@ -0,0 +1,196 @@
+            imgCacheInfo.setOnClickListener {
+                showInfoDialog(
+                    requireContext(),
+                    title = R.string.what_is_image_replication_,
+                    descrText = R.string.explanation_of_this_feature
+                )
+            }
+
+            //todo: insert meaningful explanation for index replication
+            indexReplicationInfo.setOnClickListener {
+                showInfoDialog(
+                    requireContext(),
+                    title = R.string.what_is_index_replication_,
+                    descrText = R.string.explanation_of_this_feature
+                )
+            }
+
+            resetPreferences.setOnClickListener {
+                showInfoDialog(
+                    requireContext(),
+                    title = R.string.are_you_sure_,
+                    descrText = R.string.all_preferences_will_be_reset_to_default,
+                    posButtonText = R.string.yes,
+                    negButtonText = R.string.no,
+                    posButtonCallback = { presenter.onResetPreferencesClick() }
+                )
+            }
+        }
+    }
+
+    override fun init() {
+        Log.d(SETTINGS_SCREEN, "initializing SettingsFragment")
+        (activity as MainActivity).setSelectedTab(2)
+        (activity as MainActivity).setToolbarVisibility(false)
+        (requireActivity() as MainActivity).setBottomNavigationVisibility(true)
+    }
+
+    override fun setCrashReportPreference(crashReport: CrashReport) {
+        binding.crashGroup.check(buttonFromCrashReport(crashReport))
+    }
+
+    override fun setImgCacheReplicationPref(imgReplicationPref: ImgCacheReplication) {
+        binding.imgReplicationGroup.check(buttonFromImgReplication(imgReplicationPref))
+    }
+
+    override fun setIndexReplicationPref(indexReplication: IndexReplication) {
+        binding.indexReplicationGroup.check(buttonFromIndexReplication(indexReplication))
+    }
+
+    override fun notifyUser(message: String, moreTime: Boolean) {
+        Notifications.notifyUser(context, message, moreTime)
+    }
+
+    private fun rgButtonSelected(rg: RadioGroup, checkedID: Int) {
+        rg.children.filterIsInstance(RadioButton::class.java).forEach { rb ->
+            if (rb.id != checkedID) rb.setTextColor(getColor(R.color.colorPrimary))
+            else rb.setTextColor(getColor(R.color.white))
+        }
+    }
+
+    private fun getColor(@ColorRes colorID: Int) =
+        ContextCompat.getColor(requireContext(), colorID)
+
+    private fun crashReportFromButton(buttonID: Int): CrashReport {
+        val reversed = crashReportMap.entries.associate { (k, v) -> v to k }
+        return reversed.getOrDefault(buttonID, CrashReport.NONE)
+    }
+
+    private fun buttonFromCrashReport(crashReport: CrashReport): Int {
+        return crashReportMap[crashReport]
+            ?: throw AssertionError("CrashReport must be of known type")
```

Comment:
Almost good. It won't hurt to check invariants in debug builds indeed. But `crashReportMap` has only 2 entries and `crashReport` can be 1 of 3 values, `NONE` is legal value and it will cause `AssertionError`. `AssertionError`s must be thrown only when we are sure they are never happen. This is double-check for situations when we assume something. It is helpful when we are wrong and some invariant doesn't hold.